cobra/core/Reaction.py

```diff
@@ -76,14 +78,282 @@ def __init__(self, id=None, name='', subsystem='', lower_bound=0.,
+        forward_variable = self.forward_variable
+        reverse_variable = self.reverse_variable
+        self._id = value
+        self.model.reactions._generate_index()
+        forward_variable.name = self._get_forward_id()
+        reverse_variable.name = self._get_reverse_id()
+
+    def _get_reverse_id(self):
+        """Generate the id of reverse_variable from the reaction's id."""
+        return '_'.join((self.id, 'reverse',
+                         hashlib.md5(
+                             self.id.encode('utf-8')).hexdigest()[0:5]))
+
+    def _get_forward_id(self):
+        """Generate the id of forward_variable from the reaction's id."""
+        return self.id
+
+    @property
+    def flux_expression(self):
+        """Forward flux expression
+
+        Returns
+        -------
+        sympy expression:
+            The expression represeenting the the forward flux (if associated
+            with model), otherwise None. Representing the net flux if
+            model.reversible_encoding == 'unsplit' or None if reaction is
+            not associated with a model """
+        if self.model is not None:
+            return 1. * self.forward_variable - 1. * self.reverse_variable
+        else:
+            return None
+
+    @property
+    def forward_variable(self):
+        """An optlang variable representing the forward flux
+
+        Returns
+        -------
+        optlang.interface.Variable:
+            An optlang variable for the forward flux or None if reaction is
+            not associated with a model.
+        """
+        if self.model is not None:
+            if self._forward_variable is None:
+                self._forward_variable = self.model.solver.variables[
+                    self._get_forward_id()]
+            assert self._forward_variable.problem is self.model.solver
+            return self._forward_variable
+        else:
+            return None
+
+    @property
+    def reverse_variable(self):
+        """An optlang variable representing the reverse flux
+
+        Returns
+        -------
+        optlang.interface.Variable:
+            An optlang variable for the reverse flux or None if reaction is
+            not associated with a model.
+        """
+        model = self.model
+        if model is not None:
+            if self._reverse_variable is None:
+                self._reverse_variable = model.solver.variables[
+                    self._get_reverse_id()]
+            assert self._reverse_variable.problem is self.model.solver
+            return self._reverse_variable
+        else:
+            return None
+
+    def _reset_var_cache(self):
+        self._forward_variable = None
+        self._reverse_variable = None
+
+    @property
+    def objective_coefficient(self):
+        """ Get or set the objective coefficient (float)
+        """
+        if self.model is not None and self.model.solver.objective is not None:
+            coefficients_dict = \
```

Comment:
One of the advantages of using optlang is that you can set the objective to an arbitrary expression of reactions using `model.objective = expression`. It would be possible to update the reaction coefficients in the model.objective setter, but this would require the same sympy expression parsing. For cameo did not intend reaction.objective_coefficient to be used, so we didn't want to parse the sympy expression every time the objective was changed and this getter was implemented instead to maintain cobra-compatibility.